### PR TITLE
Bugfix/agcmd 2205

### PR DESCRIPTION
**Pull request by ShurakaiSoft**
Added stronger checking of errors returning from OpenAM, so that they are not accidentally  cached in Redis.
Code cleanup.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,10 +26,9 @@
   },
   "homepage": "https://github.com/agco/passport-openam",
   "dependencies": {
-    "MD5": "^1.2.1",
-    "bluebird": "^2.9.24",
+    "bluebird": "^3.3.4",
     "http-as-promised": "^1.0.0",
-    "lodash": "^3.7.0",
+    "md5": "^2.1.0",
```
Comment (ssebro) [inline package.json:31]: these are significant changes. Are you aware of what the changes are, and what the impact is?

File: test/redisOpenAM.spec.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        it('should return an instance of BasicStrategy', function () {
+            basicStrategy.should.be.an.Object;
+            basicStrategy.should.be.an.instanceof(BasicStrategy);
+        });
+
+        describe('The verify function used by createBasicStrategy', function () {
+            var verify;
+
+            before(function getRefToVerify() {
+                verify = basicStrategy._verify;
```
Comment (blabno) [inline test/redisOpenAM.spec.js:76]: I think we should not test internals. Rather test on interface level.

File: package.json (hunk 1 above)
Comment (ShurakaiSoft) [inline package.json:31]: lodash is not used so it's safe to delete.
The version of md5 we're using has been depreciated. So I've updated to the latest version. I'm trusting that our unit tests have good coverage and will catch any issues.

File: test/redisOpenAM.spec.js (hunk 2 above)
Comment (ShurakaiSoft) [inline test/redisOpenAM.spec.js:76]: I want to unit test the verify function and this is the quickest and easiest way to do that with the least amount of mocking. However I'm open to other suggestions on testing the verify function.

File: lib/redisOpenAM.js [hunk 3]
```diff
@@ -17,116 +39,60 @@ function createBasicStrategy(options) {
...
-        }
-
-        function invalidate(err) {
-            return done(err, null);
-        }
+            .then(function returnUser(authedUser) {
+                return done(null, authedUser);
+            })
+            .catch(function invalidateAllErrors(err) {
+                console.error('[Oauth2Stragegy]:', err)
```
Comment (ssebro) [inline lib/redisOpenAM.js]: Oauth2Stragegy spelt wrong

**Pull request merged**